### PR TITLE
[SPARK-29526][SQL] UNCACHE TABLE should look up catalog/table like v2 commands 

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

Add UncacheTableStatement and make UNCACHE TABLE go through the same catalog/table resolution framework of v2 commands.

### Why are the changes needed?

It's important to make all the commands have the same table resolution behavior, to avoid confusing end-users. e.g.

```
USE my_catalog
DESC t // success and describe the table t from my_catalog
UNCACHE TABLE t // report table not found as there is no table t in the session catalog
```
### Does this PR introduce any user-facing change?

yes. When running UNCACHE TABLE, Spark fails the command if the current catalog is set to a v2 catalog, or the table name specified a v2 catalog.

### How was this patch tested?

New unit tests

### CODE REVIEW

**Comment by imback82** [comment]
cc: @cloud-fan / @viirya / @rdblue

**Comment by imback82** [comment]
@viirya When #26179 is merged, I can rearrange the code so that cache/uncache go together.

**Review by viirya — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala [hunk 1]
```diff
@@ -339,6 +339,13 @@ case class CacheTableStatement(
     isLazy: Boolean,
     options: Map[String, String]) extends ParsedStatement
 
+/**
+ * A UNCACHE TABLE statement, as parsed from SQL
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala]: `A` -> `An`?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala [hunk 2]
```diff
@@ -1062,6 +1062,16 @@ class DDLParserSuite extends AnalysisTest {
       "It is not allowed to add catalog/namespace prefix a.b")
   }
 
+  test("UNCACHE TABLE table") {
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala]: Duplicated `TABLE table`. "UNCACHE TABLE" or "UNCACHE table" would be correct.

**Review by dongjoon-hyun — APPROVED** [review]
LGTM. (I left two minor comments, @imback82 ).

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala (hunk 1 above)
Comment (imback82) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala]: Fixed, thanks.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala (hunk 2 above)
Comment (imback82) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/DDLParserSuite.scala]: Ah OK. I see some discrepancies. I fixed other test names to have command as the test name.

**Comment by dongjoon-hyun** [comment]
Thank you, @imback82 , @cloud-fan , @viirya .

The last commit is only comment and test case name changes in `DDLParserSuite`. For `DDLParserSuite`, I manually ran the test again.

Merged to master.

**Comment by imback82** [comment]
Thanks everyone!

**Pull request closed without merging**